frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx

```diff
@@ -165,6 +235,238 @@ export const CreateSubscriptionYAML: React.SFC<CreateSubscriptionYAMLProps> = (p
   );
 };
 
+const tableColumnClasses = [
+  classNames('col-lg-3', 'col-md-3', 'col-sm-4', 'col-xs-6'),
+  classNames('col-lg-2', 'col-md-3', 'col-sm-4', 'col-xs-6'),
+  classNames('col-lg-2', 'col-md-3', 'col-sm-4', 'hidden-xs'),
+  classNames('col-lg-3', 'col-md-3', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-2', 'hidden-md', 'hidden-sm', 'hidden-xs'),
+  Kebab.columnClass,
+];
+
+const CatalogSourceHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'name',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Publisher',
+      sortField: 'publisher',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Availability',
+      sortField: 'availability',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: 'Endpoint',
+      sortField: 'endpoint',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[3] },
+    },
+    {
+      title: '# of Operators',
+      sortField: 'operatorCount',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[4] },
+    },
+    {
+      title: '',
+      props: { className: tableColumnClasses[5] },
+    },
+  ];
+};
+
+const getEndpoint = (catalogSource) => {
+  if (catalogSource.spec.configmap) {
+    return (
+      <ResourceLink
+        kind={referenceForModel(ConfigMapModel)}
+        name={catalogSource.spec.configmap}
+        namespace={catalogSource.metadata.namespace}
+      />
+    );
+  }
+  return catalogSource.spec.image || catalogSource.spec.address;
+};
+
+const getOperatorCount = (catalogSource, packageManifests) =>
+  _.filter(packageManifests, {
+    status: {
+      catalogSource: catalogSource.metadata.name,
+      catalogSourceNamespace: catalogSource.metadata.namespace,
+    },
+  }).length;
+
+const CatalogSourceTableRow: React.FC<CatalogSourceTableRowProps> = ({
+  obj: {
+    availability = '-',
+    disabled = false,
+    endpoint = '-',
+    isDefault = false,
+    name,
+    operatorCount = 0,
+    operatorHub,
+    publisher = '-',
+    source,
+  },
+  index,
+  key,
+  style,
+}) => {
+  return (
+    <TableRow
+      className={disabled && 'catalog-source__table-row--disabled'}
+      id={source ? source.metadata.uid : index}
+      index={index}
+      style={style}
+      trKey={key}
+    >
+      <TableData className={tableColumnClasses[0]}>
+        {source ? (
+          <ResourceLink
+            kind={catalogSourceModelReference}
+            name={source.metadata.name}
+            namespace={source.metadata.namespace}
+            title={source.metadata.name}
+          />
+        ) : (
+          name
+        )}
+      </TableData>
+      <TableData className={tableColumnClasses[1]}>{publisher || '-'}</TableData>
```

Comment:
It looks like we're both defaulting the value above and handling empty values here. We should do one or the other.